osquery/tables/system/darwin/crash_logs.cpp

```diff
@@ -0,0 +1,144 @@
+    {"Process", "process"},
+    {"Path", "application_path"},
+    {"Identifier", "identifier"},
+    {"Version", "version"},
+    {"Parent Process", "parent_proc"},
+    {"Responsible", "responsible"},
+    {"User ID", "user_id"},
+    {"Date/Time", "date_time"},
+    {"Crashed Thread", "crashed_thread"},
+    {"Exception Type", "exception_type"},
+    {"Exception Codes", "exception_codes"},
+    {"Exception Note", "exception_notes"},
+    {"rax", "rax"},
+    {"rbx", "rbx"},
+    {"rcx", "rcx"},
+    {"rdx", "rdx"},
+    {"rdi", "rdi"},
+    {"rsi", "rsi"},
+    {"rbp", "rbp"},
+    {"rsp", "rsp"},
+};
+
+void readCrashDump(const std::string& app_log, Row& r) {
+  auto apl = fs::path(app_log);
+  r["log_location"] = app_log;
+  std::string content;
+
+  if (readFile(apl, content).ok()) {
+
+    // Variables for capturing the stack trace
+    boost::format crashed_thread_format("Thread %1% Crashed");
+    auto crashed_thread_seen = false;
+
+    auto lines = split(content, "\n");
+    for (auto it = lines.begin(); it != lines.end(); it++) {
+      auto line = *it;
+      // Tokenize each line by a colon, to get key/value pairs.
+      auto toks = split(line, ":");
+
+      // Grab the most recent stack trace line of the crashed thread.  If we so
+      // choose, we could grab the rest of the stack context here, or at least
+      // more frames.
+      if (crashed_thread_seen && toks[0] == crashed_thread_format.str()) {
+        r["stack_trace"] = *(++it);
+        crashed_thread_seen = false;
+      }
+
+      if (crashDumpKeys.count(toks[0])) {
+        // If we're looking at the registers line, we further process the line
+        // to get each register value.
+        if (toks[0] == "rax" || toks[0] == "rdi") {
+          toks = split(line, ": ");
+          for (size_t i = 0; i < toks.size(); i += 2) {
+            // Make sure that the register value we're about to grab is one of
+            // the keys we care about.
+            r[crashDumpKeys.at(toks[i])] = toks[i + 1];
+          }
+        }
+        // If the token is the date time, reconstruct the previously split
+        // upon colons.
+        else if (toks[0] == "Date/Time") {
+          r[crashDumpKeys.at(toks[0])] =
+              toks[1] + ":" + toks[2] + ":" + toks[3];
+        }
+        // If the token is the Crashed thread, update the format string so we
+        // can later grab all of the stack trace
+        else if (toks[0] == "Crashed Thread") {
+          r[crashDumpKeys.at(toks[0])] = split(toks[1], " ")[0];
+          crashed_thread_format % r[crashDumpKeys.at(toks[0])].c_str();
+          crashed_thread_seen = true;
+        } else {
+          // otherwise, process the line normally.
+          r[crashDumpKeys.at(toks[0])] = toks[1];
+        }
+      }
+    }
+  }
+}
+
+QueryData genCrashLogs(QueryContext& context) {
+  QueryData results;
+  // Process system logs
+  auto sys_logs = fs::path("/Library/Logs/DiagnosticReports");
```

Comment:
Create a `const std::string`, name it `kDiagnosticReportsPath` or something similar, and pass that directly to `listFilesInDirectory`, it will provide the implicit casting.